### PR TITLE
Use base aws classes in Amazon ECS Operators/Sensors/Triggers

### PR DESCRIPTION
**Pull request by jayceslesar**
Similar to https://github.com/apache/airflow/pull/35133 (relates to https://github.com/apache/airflow/issues/35278)

Unsure how to get tests passing for `test_hook_and_client` in `tests/providers/amazon/aws/operators/test_ecs.py`, because it seems like it should just work 🤷

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/ecs.py [hunk 1]
```diff
@@ -45,21 +46,18 @@
...
+    def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-    @cached_property
-    def hook(self) -> EcsHook:
-        """Create and return an EcsHook."""
-        return EcsHook(aws_conn_id=self.aws_conn_id, region_name=self.region)
+    @property
+    def _hook_parameters(self) -> dict[str, Any]:
+        return {**super()._hook_parameters}
```
Comment (Taragolis) [inline airflow/providers/amazon/aws/operators/ecs.py]:
```suggestion
```

Redundant for Thin-wrapped hooks

File: airflow/providers/amazon/aws/sensors/ecs.py [hunk 2]
```diff
@@ -45,18 +44,17 @@ def _check_failed(current_state, target_state, failure_states, soft_fail: bool)
...
+    def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-    @cached_property
-    def hook(self) -> EcsHook:
-        """Create and return an EcsHook."""
-        return EcsHook(aws_conn_id=self.aws_conn_id, region_name=self.region)
+    @property
+    def _hook_parameters(self) -> dict[str, Any]:
+        return {**super()._hook_parameters}
```
Comment (Taragolis) [inline airflow/providers/amazon/aws/sensors/ecs.py]:
```suggestion
```

File: airflow/providers/amazon/aws/operators/ecs.py [hunk 3]
```diff
@@ -45,21 +46,18 @@
...
-class EcsBaseOperator(BaseOperator):
+class EcsBaseOperator(AwsBaseOperator[EcsHook]):
     """This is the base operator for all Elastic Container Service operators."""
 
-    def __init__(self, *, aws_conn_id: str | None = DEFAULT_CONN_ID, region: str | None = None, **kwargs):
-        self.aws_conn_id = aws_conn_id
-        self.region = region
+    aws_hook_class = EcsHook
+
+    def __init__(self, **kwargs):
```
Comment (Taragolis) [inline airflow/providers/amazon/aws/operators/ecs.py]: I guess `__init__` constructor could be removed

File: airflow/providers/amazon/aws/sensors/ecs.py [hunk 4]
```diff
@@ -45,18 +44,17 @@ def _check_failed(current_state, target_state, failure_states, soft_fail: bool)
...
-class EcsBaseSensor(BaseSensorOperator):
+class EcsBaseSensor(AwsBaseSensor[EcsHook]):
     """Contains general sensor behavior for Elastic Container Service."""
 
-    def __init__(self, *, aws_conn_id: str | None = DEFAULT_CONN_ID, region: str | None = None, **kwargs):
-        self.aws_conn_id = aws_conn_id
-        self.region = region
+    aws_hook_class = EcsHook
+
+    def __init__(self, **kwargs):
```
Comment (Taragolis) [inline airflow/providers/amazon/aws/sensors/ecs.py]: I guess ``__init__`` constructor could be removed

**Review by Taragolis — COMMENTED** [review]
>Unsure how to get tests passing for test_hook_and_client in tests/providers/amazon/aws/operators/test_ecs.py, because it seems like it should just work 🤷

The better is rewrite, because in this case it mock different object. You need to mock `airflow.providers.amazon.aws.operators.ecs.EcsBaseOperator.aws_hook_class` however it would not work because it would fail on subclass check

You could split it by two test in first just check that expected argument are passed into the hook constructor, something similar to the 

https://github.com/apache/airflow/blob/56265904062960b681dc1d5237518b3e76b87296/tests/providers/amazon/aws/operators/test_athena.py#L80-L101

And second test just mock hook property and check and check client property

**Comment by jayceslesar** [comment]
@Taragolis thank you for the review -- all but mocking the hook property and client property should be all set. Pretty unsure how to best do that? Should I just test that the hook was not called and keep the client portion of the test the same?

**Comment by Taragolis** [comment]
It could be something like that.

```python
        with mock.patch.object(EcsBaseOperator, "hook", new_callable=mock.PropertyMock) as m:
            mocked_hook = mock.MagicMock(name="FakeHook")
            mocked_client = mock.MagicMock(name="FakeBoto3Client")
            mocked_hook.conn = mocked_client
            m.return_value = mocked_hook

            assert op.client == mocked_client
            m.assert_called_once()
```

**Comment by Taragolis** [comment]
And after that you could remove `test_hook_and_client` because it not required anymore after that

**Comment by jayceslesar** [comment]
Thank you for the example!

**Comment by Taragolis** [comment]
Need to fix [Static Checks](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst):
- you could run locally `breeze static-checks {id-of-failed-check} --all-files` e.g. `breeze static-checks ruff --all-files`
- Another option run it thought pre-commit `pre-commit run {id-of-failed-check} --all-files` e.g. `pre-commit run ruff --all-files`

You could have a look what failed in [Tests / Static checks (pull_request)](https://github.com/apache/airflow/actions/runs/7315028607/job/19939864695?pr=36393) logs

**Review by Taragolis — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**